### ISSUE TITLE
[ISSUE] databricks_dbfs_file fills up terraform statefile

### ISSUE THREAD

**Issue by al-lac** [databrickslabs/terraform-provider-databricks#149]
Adding multiple files via the `databricks_dbfs_file` resource fills up the terraform state file. After it reaches a size of 512kb all future applies will fail with the following error:

`Failed to save state: invalid character 'R' looking for beginning of value`

It should be possible to upload files to dbfs without having the content stored in the state. Would it not be possible to check if a file has changed with a md5 checksum? 

### Terraform Version

`Terraform v0.12.24`

### Affected Resource(s)

- databricks_dbfs_file

### Expected Behavior
The files should be created on the dbfs filesystem.

### Actual Behavior
The files fill up the state. Some files get copied, the others can't because of the full state file. 

### Steps to Reproduce
1. Add multiple files to the environment
1. `terraform apply`

### Important Factoids
Using Databricks running on Azure, Terraform Backend is Consul

### References
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:
- hashicorp/terraform#24394
- #136

**Comment by nfx** [comment]
should be resolved in #111

@al-lac , what is the use-case of putting files to DBFS? there's Databricks Stack CLI, that is better for versioning smaller sub-projects with lots of notebooks.

**Comment by sdebruyn** [comment]
E.g. in my case: Azure Data Factory has an activity to run jobs on Databricks. That activity requires you give the path to the JAR you want to run in DBFS.

**Comment by al-lac** [comment]
Our use-case is to automatically setup spark-monitoring for every cluster. This requires us to transfer multiple JAR files + the setup script to the DBFS Filesystem. 

Transferring those via the Databricks Provider seems logical to us. The best solution would be if monitoring would work from the get-go without having to build JAR files and setting everything up manually.

**Comment by stikkireddy** [comment]
Please review: #171

**Comment by al-lac** [comment]
Looks like this is just what we wanted. Thanks a lot! 

Will test it once it is merged into master.

**Comment by stikkireddy** [comment]
closed as pr is merged
